hydromt/models/model.py

```diff
@@ -246,32 +230,35 @@ def build(
                 }
 
         """
-        opt = opt or {}
-        opt = self._check_get_opt(opt)
-
-        # merge cli region and res arguments with opt
-        if region is not None:
-            if self._CLI_ARGS["region"] not in opt:
-                opt = {self._CLI_ARGS["region"]: {}, **opt}
-            opt[self._CLI_ARGS["region"]].update(region=region)
-
-        # then loop over other methods
-        for method in opt:
-            # if any write_* functions are present in opt, skip the final self.write()
-            if method.startswith("write_"):
-                write = False
-            kwargs = {} if opt[method] is None else opt[method]
-            self._run_log_method(method, **kwargs)
-
-        # write
-        if write:
+        steps = steps or {}
+        validate_steps(self, steps)
+
+        # steps gets preference over defaults.
+        # But put region.create at the start of the list.
+        steps = pydantic.v1.utils.deep_update(
+            steps, {"region.create": {"region": region}}
+        )
+
+        tmp = steps.pop("region.create")
+        steps = {"region.create": tmp, **steps}
+
+        for step in steps:
+            self.logger.info(f"build: {step}")
+            kwargs = steps[step] or {}
+            # Call the methods.
+            rgetattr(self, step)(**kwargs)
```

Comment:
So where I put a note earlier in ``_run_log_method`` I see the behavior indeed seems to be lost. In the log file, we want to print more than just the function name but also all the arguments even the default values that the user did not include in the yaml file. All this for reproducibility.
You could argue that if the hydromt version (if not dev) is in the log file and a copy of the yml it may be enough as well for reproducibility but we do not have it in place yet.
Copy of current log messages:
![image](https://github.com/Deltares/hydromt/assets/45457510/133c1ddf-ad3f-4c2a-a059-02c6ef1e265c)